src/language-js/printer-estree.js

```diff
@@ -1630,7 +1630,28 @@ function printPathNoParens(path, options, print, args) {
         parts.push(" ");
       }
 
-      parts.push(path.call(print, "argument"));
+      if (n.argument.comments) {
+        const shouldBreak = hasNewlineInRange(
```

Comment:
Why is this necessary? Is this too bad?

```js
// input
!!(
  /* foo */
  foo
);

// Output
!!(/* foo */ foo);
```

Reply:
Sorry, I don't familiar the Prettier philosophy yet...
(Also, I don't think it is too bad.)

Should the group isn't broken?

Reply:
@sosukesuzuki We just try not to read from the original source as much as possible to make these kind of decisions

In this case, just ignore if the source has a new line or not

Reply:
I see.
For example, should Prettier how format when receive the below code.

```js
!!(
  foo
  // comment
);
```

Reply:
> For example, should Prettier how format when receive the below code?
> 
> ```js
> !!(
>   foo
>   // comment
> );
> ```

The same way as the input. But Prettier will do that when it prints the single line comment

Reply:
Okay, I fixed following it.